src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -0,0 +1,270 @@
+ * <pre>
+ * $ java -jar checkstyle-X.XX-all.jar Main.java -b "//METHOD_DEF[./IDENT[@text='sayHello']]"
+ * CLASS_DEF -&gt; CLASS_DEF [1:0]
+ * `--OBJBLOCK -&gt; OBJBLOCK [1:18]
+ * |--METHOD_DEF -&gt; METHOD_DEF [2:4]
+ * </pre>
+ * <p>
+ * The following example demonstrates validation of methods order, so that public methods should
+ * come before the private ones:
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt;
+ *  &lt;property name="query" value="//METHOD_DEF[.//LITERAL_PRIVATE and
+ *  following-sibling::METHOD_DEF[.//LITERAL_PUBLIC]]"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void method1() { }
+ *  private void method2() { } // violation
+ *  public void method3() { }
+ *  private void method4() { } // violation
+ *  public void method5() { }
+ *  private void method6() { } // ok
+ * }
+ * </pre>
+ * <p>
+ * To violate if there are any parametrized constructors
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt;
+ *  &lt;property name="query" value="//CTOR_DEF[count(./PARAMETERS/*) &gt; 0]"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public Test(Object c) { } // violation
+ *  public Test(int a, HashMap&lt;String, Integer&gt; b) { } // violation
+ *  public Test() { } // ok
+ * }
+ * </pre>
+ * <p>
+ * To violate if method name is 'test' or 'foo'
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt;
+ *  &lt;property name="query" value="//METHOD_DEF[./IDENT[@text='test' or @text='foo']]"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void test() {} // violation
+ *  public void getName() {} // ok
+ *  public void foo() {} // violation
+ *  public void sayHello() {} // ok
+ * }
+ * </pre>
+ * <p>
+ * To violate if new instance creation was done without <b>var</b> type
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;MatchXpath&quot;&gt;
+ *  &lt;property name=&quot;query&quot; value=&quot;//VARIABLE_DEF[./ASSIGN/EXPR/LITERAL_NEW
+ *  and not(./TYPE/IDENT[@text='var'])]&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *   public void foo() {
+ *     SomeObject a = new SomeObject(); // violation
+ *     var b = new SomeObject(); // OK
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code matchxpath.match}
+ * </li>
+ * </ul>
+ *
+ * @since 8.39
+ */
+@StatelessCheck
+public class MatchXpathCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text provided by user.
+     */
+    public static final String MSG_KEY = "matchxpath.match";
+
+    /** Xpath query. */
+    private String query = "";
+
+    /** Xpath expression. */
+    private XPathExpression xpathExpression;
+
+    /**
+     * Setter to xpath query.
```

Comment:
This is why property descriptions must start with a verb in the singular tense.

Reply:
changed to `Setter to specify xpath query.`